source/geometry/Quaternion.ooc

```diff
@@ -214,16 +209,17 @@ Quaternion: cover {
 	createFromEulerAngles: static func (rotationX, rotationY, rotationZ: Float) -> This {
 		This createRotationZ(rotationZ) * This createRotationY(rotationY) * This createRotationX(rotationX)
 	}
-	createRotation: static func (angle: Float, direction: FloatPoint3D) -> This {
+	createRotation: static func (angle: Float, direction: FloatVector3D) -> This {
 		halfAngle := angle / 2.0f
 		point3DNorm := direction norm
 		if (point3DNorm != 0.0f)
 			direction /= point3DNorm
-		This new(0.0f, halfAngle * direction) exponential
+		This new(0.0f, (halfAngle * direction) toFloatPoint3D()) exponential
```

Comment:
That is my suggestion and it is included in this issue https://github.com/cogneco/ooc-kean/issues/1156, but I left it out in this PR.